### PR TITLE
Store join and renew dates - Issue/186

### PR DESCRIPTION
**Pull request by pippinsplugins**
#186

### CODE REVIEW

File: includes/class-rcp-member.php [hunk 1]
```diff
@@ -230,6 +230,106 @@ public function calculate_expiration( $force_now = false ) {
...
+		if( empty( $date ) ) {
+
+			$sub_name = rcp_get_subscription_name( $subscription_id );
+			$args     = array( 'user_id' => $this->ID, 'subscription' => $sub_name, 'order' => 'ASC', 'number' => 1 );
+			$payments = new RCP_Payments;
+			$payments = $payments->get_payments( $args );
+
+			if( $payments ) {
+				$payment = reset( $payments );
+				$date    = $payment->date;
```
Comment (mindctrl) [inline includes/class-rcp-member.php]: A note to set the joined date here to avoid payment queries during future calls.

Comment (pippinsplugins) [inline includes/class-rcp-member.php]: Done.

File: includes/class-rcp-member.php [hunk 2]
```diff
@@ -230,6 +230,107 @@ public function calculate_expiration( $force_now = false ) {
...
+
+	/**
+	 * Sets the renewed date for a member
+	 *
+	 * @access  public
+	 * @since   2.6
+	*/
+	public function set_renewed_date( $date = '' ) {
+
+		if( get_use_meta( $this->ID, '_rcp_new_subscription', true ) ) {
```
Comment (mindctrl) [inline includes/class-rcp-member.php]: @pippinsplugins this should be get_user_meta. Missing the 'r' in user.

Comment (pippinsplugins) [inline includes/class-rcp-member.php]: Fixed.

File: includes/class-rcp-member.php [hunk 3]
```diff
@@ -230,6 +230,107 @@ public function calculate_expiration( $force_now = false ) {
...
+
+		if( empty( $date ) ) {
+			$date = date( 'Y-m-d H:i:s' );
+		}
+
+		if( empty( $subscription_id ) ) {
+			$subscription_id = $this->get_subscription_id();
+		}
+
+		$ret = update_user_meta( $this->ID, 'rcp_joined_date_' . $this->get_subscription_id() );
```
Comment (mindctrl) [inline includes/class-rcp-member.php]: @pippinsplugins This throws a debug notice on new registrations due to missing argument 3 for `update_user_meta()`.

Comment (mindctrl) [inline includes/class-rcp-member.php]: Does it on upgrades too. Looks to be running too early?

**Pull request merged**